pkg/controller/common/helpers.go

```diff
@@ -593,47 +595,50 @@ func removeIgnDuplicateFilesUnitsUsers(ignConfig ign2types.Config) (ign2types.Co
 	return ignConfig, nil
 }
 
-// TranspileCoreOSConfigToIgn transpiles Fedora CoreOS config to ignition
-// internally it transpiles to Ign spec v3 config
-func TranspileCoreOSConfigToIgn(files, units []string) (*ign3types.Config, error) {
+// ConvertButaneFragmentsToIgnition takes as input butane fragments (fcos 1.3) and returns an Ignition v3.2 configuration.
+func ConvertButaneFragmentsToIgnition(files, units []string) (*ign3types.Config, error) {
 	overwrite := true
-	outConfig := ign3types.Config{}
+	buCfg := fcosbutane.Config{}
 	// Convert data to Ignition resources
 	for _, contents := range files {
-		f := new(fcctbase.File)
+		f := new(bubase.File)
 		if err := yaml.Unmarshal([]byte(contents), f); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal %q into struct: %v", contents, err)
 		}
 		f.Overwrite = &overwrite
 
-		// Add the file to the config
-		var ctCfg fcctbase.Config
-		ctCfg.Storage.Files = append(ctCfg.Storage.Files, *f)
-		ign3_0config, tSet, err := ctCfg.ToIgn3_0()
-		if err != nil {
-			return nil, fmt.Errorf("failed to transpile config to Ignition config %s\nTranslation set: %v", err, tSet)
-		}
-		ign3_2config := translate3.Translate(translate3_1.Translate(ign3_0config))
-		outConfig = ign3.Merge(outConfig, ign3_2config)
+		buCfg.Storage.Files = append(buCfg.Storage.Files, *f)
 	}
+	sort.Slice(buCfg.Storage.Files, func(i, j int) bool { return buCfg.Storage.Files[i].Path < buCfg.Storage.Files[j].Path })
```

Comment:
Why is sorting here (and below) necessary? Can you add a comment?

Reply:
Good point!  Done.  It may not actually be necessary...I hit an issue that made me think it was, but perhaps it isn't?

But hmm, I think because we md5sum the machineconfigs today, we really have to be sure they're "canonicalized".